src/analysis/souffle/examples/multimic/multimic.arcs

```diff
@@ -0,0 +1,59 @@
+// Note: This file marks speculative syntax with a SPECULATIVE comment before
+// the speculative syntax.
+
+// Represents an actual microphone within a smart mic listening to audio.
+particle MicIn
+  audio: writes AudioData { }
+// SPECULATIVE: claim a QoS integrity tag.
+//claim audio is userSpeech and is integrity#speechToTextQoS
+  claim audio is userSpeech and is speechToTextQoS
+
+// Represents the CPU within the mic.
+particle MicCompute
+  audioIn: reads AudioData { }
+  asrNeuralNet: reads writes ASRNeuralNet { }
+  asrStorageConn: writes ASRData { }
+  claim asrStorageConn is textTranscript
+//SPECULATIVE: preserve integrity tag
+//claim asrStorageConn preserves speechToTextQoS from audioIn
+
+particle SpeechToTextStorageSink
+  speechToTextIn: reads ASRData { }
+//SPECULATIVE: check integrity tag
+//check speechToTextIn is integrity#speechToTextQoS
+  check speechToTextIn is speechToTextQoS
+
+// A particle indicating that just forwards the contents of an ASR
+// neural net to another storage location.
+particle ForwardAsrNet
+  inNeuralNet: reads ASRNeuralNet { }
+  outNeuralNet: writes ASRNeuralNet { }
+
+recipe DoASR
+  m1AsrNetStorage: create
+  m1AsrStorage: create
+  MicIn
+    audio: writes micAudio
+  MicCompute
+    audioIn: reads micAudio
+    asrNeuralNet: reads writes m1AsrNetStorage
+    asrStorageConn: writes m1AsrStorage
+  SpeechToTextStorageSink
+    speechToTextIn: reads m1AsrStorage
+
+recipe HandOffASRToOtherMic
+  m2AsrNetStorage: create
+  m1AsrNetStorage: map 'm1AsrNetStorage'
+  m1AsrStorage: use 'm1AsrStorage'
+  ForwardAsrNet
+    inNeuralNet: reads m1AsrNetStorage
+    outNeuralNet: writes m2AsrNetStorage
```

Comment:
Are you saying that for hand off, `mic1` will also forward an ML model to `mic2`?

Reply:
Yes, I was thinking that the model would be continually personalized to the user(s) of `mic1`, and that exporting the model would allow `mic2` to continue with the same level of experience. But this probably isn't a necessary feature.

Reply:
Fixed.